pkg/cloudprovider/aws/utils/utils.go

```diff
@@ -15,19 +15,24 @@ limitations under the License.
 package utils
 
 import (
-	"github.com/aws/aws-sdk-go/aws"
 	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
 )
 
-// NormalizeArchitecture translates architecture into an AWS-recognized architecture name
-func NormalizeArchitecture(architecture *string) *string {
-	if architecture == nil {
-		return nil
-	}
-	switch *architecture {
-	case v1alpha1.ArchitectureAmd64:
-		return aws.String("x86_64")
-	default:
-		return architecture
+var kubeToAWSArchitectures = map[string]string{
+	v1alpha1.ArchitectureAmd64: "x86_64",
+	v1alpha1.ArchitectureArm64: v1alpha1.ArchitectureArm64,
+}
+
+// EC2Architecture translates architecture into an AWS-recognized architecture name
+func EC2Architecture(architecture string) string {
+	return kubeToAWSArchitectures[architecture]
```

Comment:
are constraints validated and cleaned? If they're not, can run into an error here.

Reply:
We use GetArchitectures in the cloud provider api to ensure this invariant.